base/utils/rpm.go

```diff
@@ -96,3 +97,32 @@ func (n Nevra) EVRAStringE(showEpoch bool) string {
 func (n Nevra) EVRAString() string {
 	return n.EVRAStringE(false)
 }
+
+func (n Nevra) IsLessVersion(other *Nevra) (bool, error) {
+	versionA := strings.Split(n.Version, ".")
```

Comment:
Comparing NEVRAs is (way) more complex and it's already implemented in (system) rpmlib so we don't need to reinvent it.

I've pushed the fix into your branch.

Reply:
Awesome! Much, much cleaner.